packages/adminrouter/extra/src/test-harness/tests/test_master.py

```diff
@@ -1,183 +1,367 @@
+             '/logs/v1/'),
+            '/system/v1/logs/v1/'),
+         EndpointTestPaths(
+            ('/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S0'
+             '/metrics/v0/'), '/system/v1/metrics/v0/'),
+         EndpointTestPaths(
+            ('/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S0'
+             '/logs/v1'), '/system/v1/logs/v1'),
+         EndpointTestPaths(
+            ('/system/v1/agent/de1baf83-c36c-4d23-9cb0-f89f596cd6ab-S0'
+             '/metrics/v0'), '/system/v1/metrics/v0'),
+         ],
+        None, "HTTP/1.1", True, 'http://127.0.0.3:61001'),
+    EndpointTestConfig(
+        [EndpointTestPaths(
+            '/system/v1/leader/mesos/foo/bar?key=value&var=num',
+            '/system/v1/foo/bar?key=value&var=num'),
+         EndpointTestPaths(
+            '/system/v1/leader/mesos/', '/system/v1/'),
+         EndpointTestPaths(
+            '/system/v1/leader/mesos', '/system/v1'),
+         EndpointTestPaths(
+            '/system/v1/leader/marathon/foo/bar?key=value&var=num',
+            '/system/v1/foo/bar?key=value&var=num'),
+         EndpointTestPaths(
+            '/system/v1/leader/marathon/', '/system/v1/'),
+         EndpointTestPaths(
+            '/system/v1/leader/marathon', '/system/v1'),
+         ],
+        None, 'HTTP/1.1', True, 'http://127.0.0.2:80'),
+    EndpointTestConfig(
+        [EndpointTestPaths(
+            '/system/v1/logs/v1/foo/bar', '/foo/bar'),
+         EndpointTestPaths(
+            '/system/v1/logs/v1/', '/'),
+         ],
+        None, 'HTTP/1.1', True, 'http:///run/dcos/dcos-log.sock'),
+    EndpointTestConfig(
+        [EndpointTestPaths(
+            '/system/v1/metrics/foo/bar', '/foo/bar'),
+         EndpointTestPaths(
+            '/system/v1/metrics/', '/'),
+         ],
+        None, 'HTTP/1.0', True, 'http:///run/dcos/dcos-metrics-master.sock'),
+]
+
+correct_upstream_req_endpoints = []
+for x in endpoint_test_configuration:
+    if x.upstream_http_ver is not None:
+        for y in x.test_paths:
+            if y.expected_path is not None:
+                e = (y.sent_path, y.expected_path, x.upstream_http_ver)
+                correct_upstream_req_endpoints.append(e)
+
+upstream_headers_endpoints = []
+for x in endpoint_test_configuration:
+    if x.upstream_headers is not None:
+        for y in x.test_paths:
+            e = (y.sent_path, x.jwt_forwarded)
+            upstream_headers_endpoints.append(e)
+
+correct_upstream_dest_endpoints = []
+for x in endpoint_test_configuration:
+    if x.correct_upstream is not None:
+        for y in x.test_paths:
+            e = (y.sent_path, x.correct_upstream)
+            correct_upstream_dest_endpoints.append(e)
+
+LocationHeaderRewriteTestConfig = collections.namedtuple(
+    "LocationHeaderRewriteTestConfig",
+    ('endpoint_id, '
+     'basepath, '
+     'redirect_testscases')
+)
+LocationHeaderRewriteTestCase = collections.namedtuple(
```

Comment:
Also could be renamed to `LocationRewriteExpectation`. If you prefer to keep existing name `Case` suffix could be removed to be consistent with other names.